miner/scroll_worker_test.go

```diff
@@ -1005,85 +1005,183 @@ func TestSealBlockAfterCliquePeriod(t *testing.T) {
 		Tracer: vm.NewStructLogger(&vm.LogConfig{EnableMemory: true, EnableReturnData: true})}, nil, nil)
 	defer chain.Stop()
 
-	// Add artificial delay to transaction processing.
-	w.beforeTxHook = func() {
-		time.Sleep(time.Duration(chainConfig.Clique.Period) * 1 * time.Second)
+	// Define 3 transactions:
+	// A --> B (nonce: 0, gas: 20)
+	tx0, _ := types.SignTx(types.NewTransaction(b.txPool.Nonce(testBankAddress), testUserAddress, big.NewInt(100000000000000000), params.TxGas, big.NewInt(20*params.InitialBaseFee), nil), types.HomesteadSigner{}, testBankKey)
+	// A --> B (nonce: 1, gas: 5)
+	tx1, _ := types.SignTx(types.NewTransaction(b.txPool.Nonce(testBankAddress)+1, testUserAddress, big.NewInt(0), params.TxGas, big.NewInt(5*params.InitialBaseFee), nil), types.HomesteadSigner{}, testBankKey)
+	// B --> A (nonce: 0, gas: 20)
+	tx2, _ := types.SignTx(types.NewTransaction(b.txPool.Nonce(testUserAddress), testBankAddress, big.NewInt(0), params.TxGas, big.NewInt(20*params.InitialBaseFee), nil), types.HomesteadSigner{}, testUserKey)
+	// B --> A (nonce: 1, gas: 20)
+	tx3, _ := types.SignTx(types.NewTransaction(b.txPool.Nonce(testUserAddress)+1, testBankAddress, big.NewInt(0), params.TxGas, big.NewInt(20*params.InitialBaseFee), nil), types.HomesteadSigner{}, testUserKey)
+	// B --> A (nonce: 2, gas: 20)
+	tx4, _ := types.SignTx(types.NewTransaction(b.txPool.Nonce(testUserAddress)+2, testBankAddress, big.NewInt(0), params.TxGas, big.NewInt(20*params.InitialBaseFee), nil), types.HomesteadSigner{}, testUserKey)
+	// A --> B (nonce: 2, gas: 5)
+	tx5, _ := types.SignTx(types.NewTransaction(b.txPool.Nonce(testBankAddress)+2, testUserAddress, big.NewInt(0), params.TxGas, big.NewInt(5*params.InitialBaseFee), nil), types.HomesteadSigner{}, testBankKey)
+
+	b.txPool.AddRemotesSync([]*types.Transaction{tx0, tx1, tx2, tx3, tx4, tx5})
+	// start building pending block
+	w.startCh <- struct{}{}
+
+	time.Sleep(time.Second)
+	pending := w.pendingBlock()
+	assert.NotNil(t, pending)
+	assert.NotEmpty(t, pending.Transactions())
```

Comment:
Are we just testing that at least 1 txs is included in the pending block?

Reply:
Yeap, it is just there to make sure RPC nodes keep building their pending blocks.
This is not a new test tho, the diff looks weird because `TestSealBlockAfterCliquePeriod` was removed.